### PR TITLE
Move limit enforcement to accepting site ownership transfer

### PR DESCRIPTION
**Pull request by zoldar**
### Changes

This PR validates new owner's limits against the new site when accepting the ownership transfer instead. They are not enforced when issuing it anymore.

* Limits check now accounts for pageviews limits as well
* A bug is fixed where existing site member is doubly counted towards team member limits when ownership is transferred to them
* Checkout on choosing new plan is made possible when user has not sites but has pending ownership transfers
* When the site uses features that user's subscription lacks, the ownership transfer can still be accepted but they are warned about loosing access

### Tests
- [x] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [ ] This change does not need a documentation update

### Dark mode
- [x] The UI has been tested both in dark and light mode
- [ ] This PR does not change the UI

### CODE REVIEW

**Review by RobertJoonas — APPROVED** [review]
Left a couple of comments inline, but looks good to me overall! 👍

File: lib/plausible/site/memberships/accept_invitation.ex [hunk 1]
```diff
@@ -17,84 +17,115 @@ defmodule Plausible.Site.Memberships.AcceptInvitation do
...
+      membership = get_or_create_owner_membership(site, user)
+      multi = add_and_transfer_ownership(site, membership, user, selfhost?)
 
-        membership = Repo.preload(changes.membership, [:site, :user])
+      case Repo.transaction(multi) do
+        {:ok, changes} ->
+          if changes[:site_locker] == {:locked, :grace_period_ended_now} do
+            user = Plausible.Users.with_subscription(changes.user)
+            Billing.SiteLocker.send_grace_period_end_email(user)
+          end
```
[the remark this answers is not in the document]
Comment (zoldar) [reply lib/plausible/site/memberships/accept_invitation.ex]: You are right, we can get rid of that. I've done it in https://github.com/plausible/analytics/pull/3612/commits/fc44ac301d990ca9580e91ade81427c96a15f917. I have also used that opportunity limit self-hosted switching solely to implementation of `Invitations.ensure_can_take_ownership` that helper in further simplifying the code.

File: lib/plausible/site/memberships/accept_invitation.ex [hunk 2]
```diff
@@ -17,84 +17,115 @@ defmodule Plausible.Site.Memberships.AcceptInvitation do
...
-        {:error, _operation, error, _changes} ->
-          {:error, error}
-      end
+  defp finalize_invitation(multi, invitation) do
+    case Repo.transaction(multi) do
+      {:ok, changes} ->
+        if changes[:site_locker] == {:locked, :grace_period_ended_now} do
+          user = Plausible.Users.with_subscription(changes.user)
+          Billing.SiteLocker.send_grace_period_end_email(user)
+        end
```
[the remark this answers is not in the document]
Comment (zoldar) [reply lib/plausible/site/memberships/accept_invitation.ex]: https://github.com/plausible/analytics/pull/3612/commits/fc44ac301d990ca9580e91ade81427c96a15f917

File: lib/plausible/billing/quota.ex [hunk 3]
```diff
@@ -334,24 +365,28 @@ defmodule Plausible.Billing.Quota do
...
+  def ensure_within_plan_limits(_, _, _), do: :ok
 
-  defp exceeded_limits(%User{} = user, %Plan{} = plan, usage) do
+  defp exceeded_limits(%User{} = user, plan, usage) do
     for {limit, exceeded?} <- [
           {:team_member_limit, not within_limit?(usage.team_members, plan.team_member_limit)},
           {:site_limit, not within_limit?(usage.sites, plan.site_limit)},
-          {:monthly_pageview_limit, exceeds_monthly_pageview_limit?(user, plan, usage)}
+          {:monthly_pageview_limit,
+           exceeds_monthly_pageview_limit?(user, plan, usage.monthly_pageviews)}
```
[the remark this answers is not in the document]
Comment (zoldar) [reply lib/plausible/billing/quota.ex:389]: Okay, makes sense 👍  From what I can tell though, it's already accounted for via https://github.com/plausible/analytics/pull/3612/files#diff-d2043c5a772d701448fe604a1e26bf5575988809f147ff5be5a195675ea75164R396-R398 - right?

[the remark this answers is not in the document]
Comment (zoldar) [reply lib/plausible/billing/quota.ex:389]: In any case, I have added a test case for when this switch is set at https://github.com/plausible/analytics/pull/3612/commits/410970cd45b6ac76894fd6bf4e3288cec8421227

**Pull request merged**